ci/jenkins/scale-agent.sh

```diff
@@ -0,0 +1,226 @@
+        fi
+    done
+
+    if [ "$IP_READY" = false ]; then
+        echo "External IP for $NEW_AGENT_NAME did not become ready within 3 minutes. Exiting..."
+        return 1
+    fi
+
+
+    JOIN_COMMAND="java -jar $SWARM_CLIENT_JAR -master $JENKINS_URL -username $JENKINS_USER -password $JENKINS_API_TOKEN \
+        -name $NEW_AGENT_NAME -labels $LABEL -executors 1 -retry 3 -mode exclusive -disableSslVerification -deleteExistingClients -workDir $WORKDIR &"
+
+    # Run from agent command line
+    gcloud compute ssh "$NEW_AGENT_NAME" --zone="$ZONE" --command \
+        "wget https://repo.jenkins-ci.org/releases/org/jenkins-ci/plugins/swarm-client/3.43/swarm-client-3.43.jar -O swarm-client.jar && sudo apt-get install openjdk-11-jdk -y && $JOIN_COMMAND"
+    
+    echo "Agent $NEW_AGENT_NAME has been added."
+}
+
+function remove_agent {
+    AGENT_LIST=$(curl -s -u "$JENKINS_USER:$JENKINS_API_TOKEN" "$JENKINS_URL/computer/api/json" | jq -r '.computer[] | .displayName')
+
+    if [ -z "$AGENT_LIST" ]; then
+        echoerr "No agents found."
+        return 1
+    fi
+    
+    DELETED_ONCE=false
+    for AGENT_NAME in $AGENT_LIST; do
+        if [[ "$AGENT_NAME" == *"$AGENT_NAME_PATTERN"* ]]; then
+            echo "Checking Agent: $AGENT_NAME"
+
+            RUNNING_JOBS=$(curl -s -u "$JENKINS_USER:$JENKINS_API_TOKEN" "$JENKINS_URL/computer/$AGENT_NAME/api/json" | jq '[.executors[] | select(.currentExecutable != null)] | length')
+
+            if [ "$RUNNING_JOBS" -eq 0 ]; then
+                echo "Agent $AGENT_NAME can be safely removed"
+
+                echo "Removing label from agent $AGENT_NAME"
+                curl -X POST "$JENKINS_URL/computer/$AGENT_NAME/label" --user "$JENKINS_USER:$JENKINS_API_TOKEN" -d "labels="
+
+                # Recheck if there are no running jobs
+                RUNNING_JOBS=$(curl -s -u "$JENKINS_USER:$JENKINS_API_TOKEN" "$JENKINS_URL/computer/$AGENT_NAME/api/json" | jq '[.executors[] | select(.currentExecutable != null)] | length')
+            
+                if [ "$RUNNING_JOBS" -eq 0 ]; then
+                    # Remove agent by calling jenkins api
+                    curl -X POST "$JENKINS_URL/computer/doDelete?name=$AGENT_NAME" --user "$JENKINS_USER:$JENKINS_API_TOKEN"
+            
+                    # destroy the cloud VM
+                    sudo gcloud compute instances delete "$AGENT_NAME" --zone="$ZONE" --quiet
+                    echo "Agent $AGENT_NAME has been deleted."
+                    DELETED_ONCE=true
+                    break
+                else
+                    echo "Agent $AGENT_NAME still has $RUNNING_JOBS jobs running, cannot be deleted."
+                fi
+            else 
+                echo "Agent $AGENT_NAME is running $RUNNING_JOBS jobs, cannot be deleted."
+            fi
+        fi
+    done
+
+    if [ "$DELETED_ONCE" = false ]; then
+        echo "No agents can be removed."
+    fi
+}
+
+if [[ "$RUN_SETUP_ONLY" != true ]]; then
```

Comment:
By default RUN_SETUP_ONLY and RUN_CLEANUP_ONLY are false, when scale-agent.sh is called without parameters, will this logic cause it runs only remove_agent then exit?

It seems the logic should be like:

```
if RUN_SETUP_ONLY -> add_agent, exit 0

if RUN_CLEANUP_ONLY  -> remove_agent, exit 0

remove_agent
add_agent
```